### ISSUE TITLE
Error [WinError 10054]

### ISSUE THREAD

**Issue by vstadnytskyi-FDA** [caproto/caproto#789]
I have a caproto server running on Windows 10. On the same machine I am trying to access data via threading client - see below.

I get an error -> [WinError 10054]

Any clue why?
```
In [1]: import logging
   ...: from logging import debug, info, warning, error
   ...:
   ...: logging.getLogger(__name__).addHandler(logging.NullHandler())
   ...:
   ...: from caproto.threading.client import Context
   ...:
   ...: ctx = Context()
   ...: prefix = "camera_FDA"
   ...: image, frameID = ctx.get_pvs(f"{prefix}:image", f"{prefix}:frameID")
Removing <caproto.threading.client.SharedBroadcaster object at 0x00000205892809D0> due to [WinError 10054] An existing connection was forcibly closed by the remote host (10054)
```

**Comment by klauer** [comment]
Can you do a `caproto-get -vvv` with the image/frameID and let us see how it's being discovered?

**Comment by vstadnytskyi-FDA** [comment]
Here is what I am getting on the server side
```

DEBUG:caproto.bcast:VersionRequest(priority=0, version=13)
DEBUG:caproto.bcast:SearchRequest(name='camera_FDA:image', cid=41928, version=13, reply=5)
DEBUG:caproto.bcast:SearchRequest(name='camera_FDA:frameID', cid=41929, version=13, reply=5)
DEBUG:caproto.bcast:VersionResponse(version=13)
DEBUG:caproto.bcast.search:SearchResponse(port=5064, ip='255.255.255.255', cid=41928, version=13)
DEBUG:caproto.bcast.search:SearchResponse(port=5064, ip='255.255.255.255', cid=41929, version=13)
DEBUG:caproto.bcast:VersionRequest(priority=0, version=13)
DEBUG:caproto.bcast:SearchRequest(name='camera_FDA:image', cid=41928, version=13, reply=5)
DEBUG:caproto.bcast:SearchRequest(name='camera_FDA:frameID', cid=41929, version=13, reply=5)
DEBUG:caproto.bcast:VersionResponse(version=13)
DEBUG:caproto.bcast.search:SearchResponse(port=5064, ip='255.255.255.255', cid=41928, version=13)
DEBUG:caproto.bcast.search:SearchResponse(port=5064, ip='255.255.255.255', cid=41929, version=13)
DEBUG:caproto.bcast:VersionRequest(priority=0, version=13)
DEBUG:caproto.bcast:SearchRequest(name='camera_FDA:image', cid=26009, version=13, reply=5)
DEBUG:caproto.bcast:SearchRequest(name='camera_FDA:frameID', cid=26010, version=13, reply=5)
DEBUG:caproto.bcast:VersionResponse(version=13)
DEBUG:caproto.bcast.search:SearchResponse(port=5064, ip='255.255.255.255', cid=26009, version=13)
DEBUG:caproto.bcast.search:SearchResponse(port=5064, ip='255.255.255.255', cid=26010, version=13)
DEBUG:caproto.bcast:Beacon(version=13, server_port=5064, beacon_id=48, address='0.0.0.0')
DEBUG:caproto.bcast:VersionRequest(priority=0, version=13)
DEBUG:caproto.bcast:SearchRequest(name='camera_FDA:image', cid=41928, version=13, reply=5)
DEBUG:caproto.bcast:SearchRequest(name='camera_FDA:frameID', cid=41929, version=13, reply=5)
DEBUG:caproto.bcast:VersionResponse(version=13)
DEBUG:caproto.bcast.search:SearchResponse(port=5064, ip='255.255.255.255', cid=41928, version=13)
DEBUG:caproto.bcast.search:SearchResponse(port=5064, ip='255.255.255.255', cid=41929, version=13)
DEBUG:caproto.bcast:VersionRequest(priority=0, version=13)
DEBUG:caproto.bcast:SearchRequest(name='camera_FDA:image', cid=41928, version=13, reply=5)
DEBUG:caproto.bcast:SearchRequest(name='camera_FDA:frameID', cid=41929, version=13, reply=5)
DEBUG:caproto.bcast:VersionResponse(version=13)
DEBUG:caproto.bcast.search:SearchResponse(port=5064, ip='255.255.255.255', cid=41928, version=13)
DEBUG:caproto.bcast.search:SearchResponse(port=5064, ip='255.255.255.255', cid=41929, v
```

**Comment by vstadnytskyi-FDA** [comment]
```
>caproto-get -vvv camera_FDA.image
[D 17:12:54.183     repeater:  221] Another repeater is already running; will not spawn one.
[D 17:12:54.183       client:   59] Registering with the Channel Access repeater.
[D 17:12:54.183 _broadcaster:   94] (1 of 1) RepeaterRegisterRequest(client_address='0.0.0.0')
[D 17:12:54.183       _utils:  535] Netifaces unavailable; using 127.0.0.1 as local address
[D 17:12:54.183       client:   74] Searching for 'camera_FDA.image'....
[D 17:12:54.183 _broadcaster:   94] (1 of 2) VersionRequest(priority=0, version=13)
[D 17:12:54.183 _broadcaster:   92] (2 of 2) SearchRequest(name='camera_FDA.image', cid=6014, version=13, reply=5)
[D 17:12:54.183       client:   88] 0.0.0.0:50222 --->>> 255.255.255.255:5064 2 commands 56B
[D 17:12:54.183 _broadcaster:  136] 0.0.0.0:50222 <<<--- 127.0.0.1:5065 16B RepeaterConfirmResponse(repeater_address='127.0.0.1')
[D 17:12:55.190       client:   88] 0.0.0.0:50222 --->>> 255.255.255.255:5064 1 commands 56B
[D 17:12:56.190       client:   88] 0.0.0.0:50222 --->>> 255.255.255.255:5064 1 commands 56B
Traceback (most recent call last):
  File "C:\Users\AR-VR lab W1\AppData\Local\Programs\Python\Python38\lib\site-packages\caproto\sync\client.py", line 123, in search
    bytes_received, address = udp_sock.recvfrom(ca.MAX_UDP_RECV)
socket.timeout: timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\AR-VR lab W1\AppData\Local\Programs\Python\Python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\AR-VR lab W1\AppData\Local\Programs\Python\Python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\AR-VR lab W1\AppData\Local\Programs\Python\Python38\Scripts\caproto-get.exe\__main__.py", line 7, in <module>
  File "C:\Users\AR-VR lab W1\AppData\Local\Programs\Python\Python38\lib\site-packages\caproto\commandline\get.py", line 172, in main
    response = read(pv_name=pv_name,
  File "C:\Users\AR-VR lab W1\AppData\Local\Programs\Python\Python38\lib\site-packages\caproto\sync\client.py", line 353, in read
    chan = make_channel(pv_name, udp_sock, priority, timeout)
  File "C:\Users\AR-VR lab W1\AppData\Local\Programs\Python\Python38\lib\site-packages\caproto\sync\client.py", line 160, in make_channel
    address = search(pv_name, udp_sock, timeout)
  File "C:\Users\AR-VR lab W1\AppData\Local\Programs\Python\Python38\lib\site-packages\caproto\sync\client.py", line 138, in search
    check_timeout()
  File "C:\Users\AR-VR lab W1\AppData\Local\Programs\Python\Python38\lib\site-packages\caproto\sync\client.py", line 105, in check_timeout
    raise CaprotoTimeoutError(f"Timed out while awaiting a response "
caproto._utils.CaprotoTimeoutError: Timed out while awaiting a response from the search for 'camera_FDA.image'. Search requests were sent to this address list: ['255.255.255.255'].

```

**Comment by klauer** [comment]
I think you may have mistyped the PV name, unless you changed it between the captured outputs you did (`camera:image` vs `camera_FDA.image`)

I'd recommend installing `netifaces` as it more reliably determines information about your environment and can choose better default settings for broadcast requests and such.

**Comment by vstadnytskyi-FDA** [comment]
> I think you may have mistyped the PV name, unless you changed it between the captured outputs you did (`camera:image` vs `camera_FDA.image`) - specifically the `.` vs the `:`
> 
> I'd recommend installing `netifaces` as it more reliably determines information about your environment and can choose better default settings for broadcast requests and such.

I think PV name is correct

[I 17:22:00.221       server:  183] PVs available:
    camera_FDA:image
    camera_FDA:timestamp_lab
    camera_FDA:timestamp_camera
    camera_FDA:frameID
    camera_FDA:exposure_time

**Comment by vstadnytskyi-FDA** [comment]
It makes no sense. Now it is working. I will try to reproduce the error or maybe indeed I had a typo in the name.

**Comment by vstadnytskyi-FDA** [comment]
I will close it for now. I am not sure how to reproduce the error. I guess it was indeed wrong name of the PV I was trying to access.

**Comment by vstadnytskyi** [comment]
I was able to reproduce the problem.

CAproto server code

```

from logging import debug,info,warning,error
from caproto.server import pvproperty, PVGroup, SubGroup, ioc_arg_parser, run
from numpy import nan, zeros, int16, random

class Server(PVGroup):
    
    image = pvproperty(value=zeros((3,100,100)).flatten(), dtype = int, max_length = 4000000)
    dt = pvproperty(value = 1.0)

    @image.startup
    async def image(self, instance, async_lib):
        while True:
            new_arr = random.randint(256, size=(3, 100, 100)).flatten()
            await self.image.write(value=new_arr)
            await async_lib.library.sleep(self.dt.value)

if __name__ == '__main__':
    prefix = 'BITMAP_IMAGE:'
    from pdb import pm
    from tempfile import gettempdir
    import logging
    print(gettempdir()+'/{}.log'.format(prefix))
    logging.basicConfig(filename=gettempdir()+'/{}.log'.format(prefix),
                        level=logging.INFO, format="%(asctime)s %(levelname)s: %(message)s")

    ioc_options, run_options = ioc_arg_parser(
        default_prefix=prefix,
        desc='description')

    io_server  = Server(**ioc_options)

    #start async caproto server IO
    run(io_server.pvdb, **run_options)
```

and caproto threading client code

```
#!/usr/bin/env python3
from caproto.threading.client import Context
prefix='BITMAP_IMAGE:'
ctx = Context()
pv_name = prefix+'dt'
print(f'reading PV: {pv_name}')
dt, = ctx.get_pvs(pv_name)
pv_name = prefix+'image'
print(f'reading PV: {pv_name}')
image, = ctx.get_pvs(pv_name)
```

**Comment by vstadnytskyi-FDA** [comment]
```
In [4]: sys.version_info
Out[4]: sys.version_info(major=3, minor=8, micro=8, releaselevel='candidate', serial=1)

In [6]: platform.platform()
Out[6]: 'Windows-10-10.0.19041-SP0'

In [8]: caproto.__version__
Out[8]: '0.8.1'
```

**Comment by vstadnytskyi-FDA** [comment]
I just noticed. my terminal client connects to a different caproto server that is running on the same machine but contains different set of PVs.

**Comment by klauer** [comment]
Have you by chance run into this recently, @vstadnytskyi-FDA ? If not, I'd like to assume the linked PR resolved it (#794 part of v1.0.0).

**Comment by vstadnytskyi-FDA** [comment]
> Have you by chance run into this recently, @vstadnytskyi-FDA ? If not, I'd like to assume the linked PR resolved it (#794 part of v1.0.0).

I haven't seen this error recently. I think it is safe to assume it was resolved. in the linked PR resolved it (https://github.com/caproto/caproto/pull/794 part of v1.0.0).
